cfme/tests/containers/test_chargeback_for_images.py

```diff
@@ -83,33 +57,606 @@ def assign_compute_custom_rate(new_chargeback_fixed_rate, provider):
+    report.delete()
+
+
+@pytest.yield_fixture(scope='module')
+def chargeback_report_for_daily_fixed_rate(assign_compute_daily_fixed_rate, provider):
+    # Chargeback report for daily fixed compute rate
+    report = gen_report_base(provider, assign_compute_daily_fixed_rate, 'Daily')
     yield list(report.get_saved_reports()[0].data)
     report.delete()
 
 
+@pytest.yield_fixture(scope='module')
+def chargeback_report_for_daily_variable_rate(assign_compute_daily_variable_rate, provider):
+    # Chargeback report for daily variable compute rate
+    report = gen_report_base(provider, assign_compute_daily_variable_rate, 'Daily')
+    yield list(report.get_saved_reports()[0].data)
+    report.delete()
+
+
+@pytest.yield_fixture(scope='module')
+def chargeback_report_for_weekly_fixed_rate(assign_compute_weekly_fixed_rate, provider):
+    # Chargeback report for weekly fixed compute rate
+    report = gen_report_base(provider, assign_compute_weekly_fixed_rate, 'Weekly')
+    yield list(report.get_saved_reports()[0].data)
+    report.delete()
+
+
+@pytest.yield_fixture(scope='module')
+def chargeback_report_for_weekly_variable_rate(assign_compute_weekly_variable_rate, provider):
+    # Chargeback report for weekly variable compute rate
+    report = gen_report_base(provider, assign_compute_weekly_variable_rate, 'Weekly')
+    yield list(report.get_saved_reports()[0].data)
+    report.delete()
+
+
+@pytest.yield_fixture(scope='module')
+def chargeback_report_for_monthly_fixed_rate(assign_compute_monthly_fixed_rate, provider):
+    # Chargeback report for monthly fixed compute rate
+    report = gen_report_base(provider, assign_compute_monthly_fixed_rate, 'Monthly')
+    yield list(report.get_saved_reports()[0].data)
+    report.delete()
+
+
+@pytest.yield_fixture(scope='module')
+def chargeback_report_for_monthly_variable_rate(assign_compute_monthly_variable_rate, provider):
+    # Chargeback report for monthly variable compute rate
+    report = gen_report_base(provider, assign_compute_monthly_variable_rate, 'Monthly')
+    yield list(report.get_saved_reports()[0].data)
+    report.delete()
+
+
+def abstract_test_chargeback_cost(report_data, cb_rate, rate_key, rate_interval, soft_assert):
+
+    """This is an abstract test function for test fixed rate costs.
+    It's comparing the expected value that calculated by the rate
+    to the value in the chargeback report
+    Args:
+        :py:type:`list` report_data: The report data (rows as list).
+        :py:class:`ComputeRate` cb_rate: The chargeback rate object.
+        :py:type:`str` rate_key: The rate key as it appear in the CHARGEBACK_HEADER_NAMES keys.
+        :py:type:`str` rate_interval: The rate interval, (Hourly/Daily/Weekly/Monthly).
+        :var soft_assert: soft_assert fixture.
+    """
+    report_headers = CHARGEBACK_HEADER_NAMES[rate_key]
+    interval_factor = {'Hourly': 24, 'Daily': 7, 'Weekly': 4.29, 'Monthly': 1}[rate_interval]
```

Comment:
**OPTIONAL**: Why not move this dict out into the beginning of the file. As it is we recreat ethe dictionary each time we run the function. Whilst it's only tiny, it's still unnecessary.